### ISSUE TITLE
Certificate revocation error

### ISSUE THREAD

**Issue by serpe8742** [jstedfast/MailKit#876]
Hi, I use MailKit and MimeKit components in a commercial application. I noticed that, by downloading emails via IMAP, the Sectigo certificates, I attach an example, produce a validation error. I believe this is a problem in the interpretation of the same. A workaround can be setting the CheckCertificateRevocation property to false. I think it's need a correction. 
For us is urgent.
Thank you.

I enclose an example of source code that produces the error, including commentary, the instruction that circumvents the problem.

ImapClient imapClient = new ImapClient(MailKitLogger);
				//This is workaround
                //imapClient.CheckCertificateRevocation = false;
                //End workaround
                imapClient.Timeout = 20000;
                imapClient.Connect("imaps.pec.aruba.it", 993, true);
                imapClient.AuthenticationMechanisms.Remove("XOAUTH2");
                object s = "";
                if (InputDialog.Show("Insert password", "", ref s, inputDialog1.Use2Columns) == DialogResult.OK)
                {                    
                    imapClient.Authenticate("mail@pec.it", s.ToString());                   
                    imapClient.SslProtocols = System.Security.Authentication.SslProtocols.Ssl2;
                    if (imapClient.IsConnected)
                    {                      
                        var personal = imapClient.GetFolders(imapClient.PersonalNamespaces[0]);
                        foreach (MailFolder folder in personal)
                        {
                            MessageBox.Show(string.Format("[folder] {0}", folder.Name));
                        }                       
                        imapClient.Inbox.Open(FolderAccess.ReadOnly);
                        if (imapClient.Capabilities.HasFlag(ImapCapabilities.Sort))
                        {
                            var orderBy = new[] { OrderBy.ReverseArrival };
                            foreach (var uid in imapClient.Inbox.Sort(SearchQuery.All, orderBy))
                            {
                                var message = imapClient.Inbox.GetMessage(uid);
                                MessageBox.Show(string.Format("[match] {0}: {1}", uid, message.Subject));
                            }
                        }
                        else
                        {
                            // handle this in another way
                            var reversed = new UniqueIdSet(MailKit.Search.SortOrder.Descending);
                            foreach (var uid in imapClient.Inbox.Search(SearchQuery.All))
                                reversed.Add(uid);
                            foreach (var uid in reversed)
                            {
                                var message = imapClient.Inbox.GetMessage(uid);
                                MessageBox.Show(string.Format("[match] {0}: {1}", uid, message.Subject));
                            }
                        }
                        imapClient.Disconnect(true);
                    }
                }

 - OS: Windows10 / Windows 7
 - Browser Firefox/Chrome/Edge
 - Version 2.2.0 (Mailki/Mimekit)
[Sectigo.zip](https://github.com/jstedfast/MailKit/files/3367000/Sectigo.zip)

**Comment by serpe8742** [comment]
This morning the app works properly, without workaround.
I suppose the provider corrected something.
I believe we can close the report without further intervention.
Sorry.

**Comment by jstedfast** [comment]
CheckCertificateRevocation is an option passed to .NET’s SslStream telling it whether or not it should check the revocation status of the certificates. If the CA’s server is offline, then obviously revocation checks will fail because it will be unreachable. That’s what happened to you.

**Comment by serpe8742** [comment]
Thanks for the explanation.

**Comment by jstedfast** [comment]
No problem

**Comment by HubKing** [comment]
@JamesSkemp How can I disable revocation check? The mail server is my private server and it is using a certificate signed by a CA that was generated by me and has been added to the Windows certificate store. I don't know how "revocation check" is done, but if it as querying some sort of servers on the Internet, of course it would return error, because the CA does not exist. I made the certificates using XCA, so I do not have any server for certificates within my network.

MainKit could not correct to the server due to certificate error, and the error seemed to be revocation check.

![image](https://user-images.githubusercontent.com/10900489/202900868-df6d44b4-5cbc-4dfe-8cfe-0c989b7ecef9.png)

I tried calling `ServicePointManager.CheckCertificateRevocationList = false` before creating new ImapClient, but that did not work. Is there a way to disable revocation check?
